ext/opentelemetry-ext-celery/src/opentelemetry/ext/celery/__init__.py

```diff
@@ -0,0 +1,251 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Instrument `celery`_ to report Celery APP operations.
+
+There are two options for instrumenting code. The first option is to use the
+``opentelemetry-auto-instrumentation`` executable which will automatically
+instrument your Celery APP. The second is to programmatically enable
+instrumentation as explained in the following section.
+
+.. _celery: https://pypi.org/project/celery/
+
+Usage
+-----
+
+Be sure rabbitmq is running:
+
+.. code::
+
+    docker run -p 5672:5672 rabbitmq
+
+.. code:: python
+
+    from opentelemetry import trace
+    from opentelemetry.ext.celery import CeleryInstrumentor
+
+    CeleryInstrumentor().instrument()
+
+    from celery import Celery
+
+    app = Celery("tasks", broker="amqp://localhost")
+
+    @app.task
+    def add(x, y):
+        return x + y
+
+    add.delay(42, 50)
+
+API
+---
+"""
+
+import logging
+import signal
+
+from celery import registry, signals  # pylint: disable=no-name-in-module
+
+from opentelemetry import trace
+from opentelemetry.auto_instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.ext.celery.utils import (
+    attach_span,
+    detach_span,
+    retrieve_span,
+    retrieve_task_id,
+    set_attributes_from_context,
+)
+from opentelemetry.ext.celery.version import __version__
+from opentelemetry.trace.status import Status, StatusCanonicalCode
+
+logger = logging.getLogger(__name__)
+
+# Task operations
+_TASK_TAG_KEY = "celery.action"
+_TASK_APPLY_ASYNC = "apply_async"
+_TASK_RUN = "run"
+
+_TASK_RETRY_REASON_KEY = "celery.retry.reason"
+_TASK_REVOKED_REASON_KEY = "celery.revoked.reason"
+_TASK_REVOKED_TERMINATED_SIGNAL_KEY = "celery.terminated.signal"
+_TASK_NAME_KEY = "celery.task_name"
+_MESSAGE_ID_ATTRIBUTE_NAME = "messaging.message_id"
+
+
+class CeleryInstrumentor(BaseInstrumentor):
+    # pylint: disable=unused-argument
+    def _instrument(self, **kwargs):
```

Comment:
Using pylint disable in this place disables the check for all next lines I think, docs:
http://pylint.pycqa.org/en/latest/user_guide/message-control.html

If this is used only because of `kwargs` then wouldn't it be reasonable to add `args` and `kwargs` to pylintrc as ignored? WDYT?

```
# Argument names that match this expression will be ignored. Default to name
# with leading underscore.
ignored-argument-names=_.*|^ignored_|^unused_|^kwargs|^args|^mock_.+
```